File: api/common/src/main/scala/de/htwg/zeta/common/models/modelDefinitions/concept/elements/Method.scala
package de.htwg.zeta.common.models.modelDefinitions.concept.elements

import scala.collection.immutable.ListMap
import scala.collection.immutable.Seq


case class Method(
    name: String,
    parameters: ListMap[String, AttributeType],
    description: String,
    returnType: AttributeType,
    code: String
)

object Method {

  trait MethodMap {

    val methods: Seq[Method]

    /** Methods mapped to their own names. */
    final val methodMap: Map[String, Method] = methods.map(method => (method.name, method)).toMap

  }

}

File: frontend/src/components/zetalayout/model/model-editor/model/methods/Method.js
export class Method {

    constructor(data) {
        this.name = (data && data.name) || "default"
        this.parameters = (data && data.parameters) || []
        this.description = (data && data.description) || ""
        this.returnType = (data && data.returnType) || ""
        this.code = (data && data.code) || ""
        this.value = (data && data.value) || ""
    }

    clone() {
        return new Method({
            name: this.name,
            parameters: this.parameters,
            description: this.description,
            returnType: this.returnType,
            code: this.code,
            value: this.value
        });
    }
}

